packages/block-library/src/image/image.js

```diff
@@ -412,19 +412,23 @@ export default function Image( {
 				return {};
 			}
 
-			const { getBlockBindingsSource } = unlock(
-				select( blockEditorStore )
-			);
+			const { getBlockBindingsSource, getBlockParentsByBlockName } =
+				unlock( select( blockEditorStore ) );
 			const {
 				url: urlBinding,
 				alt: altBinding,
 				title: titleBinding,
 			} = metadata?.bindings || {};
+			const hasParentPattern =
+				getBlockParentsByBlockName( clientId, 'core/block' ).length > 0;
 			return {
 				lockUrlControls:
-					!! urlBinding &&
-					getBlockBindingsSource( urlBinding?.source )
-						?.lockAttributesEditing === true,
+					( !! urlBinding &&
+						getBlockBindingsSource( urlBinding?.source )
+							?.lockAttributesEditing === true ) ||
+					// Disable editing the link of the URL if the image is inside a pattern instance.
+					// This is a temporary solution until we support overriding the link on the frontend.
+					hasParentPattern,
```

Comment:
I think it should be in a separate property called `lockHrefControls`, as the `url` attribute is for image block's `src` on the `<img>` tag (which is confusing).

The attribute name for the `<a>` tag's `href` is also `href` (not as confusing).

The block bindings code also seems wrong for this, which is I think what has caused confusion. , you can see it's incorrectly used to lock controls for the `href` (which is not a supported block binding attribute):
https://github.com/WordPress/gutenberg/blob/77878eabe77e80e3391c3980e02153c029004909/packages/block-library/src/image/image.js#L445-L459

(cc @SantosGuillamot @artemiomorales @michalczaplinski)

Reply:
This is true. I noticed this as well, but forgot to mention it after the refactor. `href` and `url` are different things in the image block 😅.

Reply:
Changed in [34df026](https://github.com/WordPress/gutenberg/pull/58660/commits/34df0261aaef55722b1cd6aea8c399cc1c8acb0c). Not sure if we need the `hrefBinding` part though 🤔.

Reply:
I don't think so as it's not a supported binding attribute.

But this should fix the issue, though I would appreciate some testing from block binding contributors, as there might have been a reason that it was like that (though I can't see one).

Reply:
I hadn't seen this comment, sorry. I left a similar comment [here](https://github.com/WordPress/gutenberg/pull/58660#discussion_r1481270628).

We locked the `href` controls when the URL is bound because it doesn't make sense to link to the attachment page or the media URL. It is not going to work as expected. It's true that maybe we can try to disable only those instead of the whole `href` controls.